lxd/apparmor/instance_forkproxy.go

```diff
@@ -52,6 +52,9 @@ profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {
   ptrace (read),
   ptrace (trace),
 
+  /etc/machine-id r,
```

Comment:
[man 5 machine-id](https://www.freedesktop.org/software/systemd/man/machine-id.html) says:

> the value in /etc/machine-id will be used. If this file is empty or missing, systemd will attempt to use the D-Bus machine ID from /var/lib/dbus/machine-id, the value of the kernel command line option container_uuid, the KVM DMI product_uuid or the devicetree vm,uuid (on KVM systems), and finally a randomly generated UUID.

So I think that allowing the read of `/etc/machine-id`, will prevent the fallback behaviour of trying to read the `product_uuid`. That said, I'm fine with the denial rule you put in :)